### PR TITLE
Add documentation for Consultation

### LINKED ISSUE #100: Write documentation for Consult

### PR DESCRIPTION
**Pull request by yhanyi**
Fixes #100 
Drafted documentation for Consultation on DeveloperGuide.md
- Adds Proposed: Consultation Management, with Architecture, Implementation and Design Considerations
- Adds two sample Use Cases for Consultation
- Update User Stories

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -410,6 +504,40 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+* 1b. Date/time already booked
+    * 1b1. System shows conflict error
+    * 1b2. Use case ends
+
+**<u>Use case: UC7 - Add Student to Consultation</u>**
+
+**MSS**
+
+1. TA lists existing consultations
```
Comment (Clarenceeey) [inline docs/DeveloperGuide.md:540]: Can add what if theres no existing consultations/no consultations that are not over yet

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -410,6 +504,40 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+2. TA selects consultation
+3. TA adds student
```
Comment (Clarenceeey) [inline docs/DeveloperGuide.md:543]: Can add what happens if the student the TA wants to add does not exist or is already in the consultation

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -410,6 +504,40 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
   * 2a1. TAHub shows an error message stating that the index is invalid.
   * 2a2. Use case ends.
 
+**<u>Use case: UC6 - Create a Consultation</u>**
+
+**MSS**
+
+1. User enters: `addconsult d/2024-10-20 t/14:00`
```
Comment (Clarenceeey) [inline docs/DeveloperGuide.md:518]: Do we also need to check if the date and time are before or after the current date and time? i.e. adding a consultation for May when it is already June.

**Review by Clarenceeey — COMMENTED** [review]
Some more extensions for the MSS.

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -239,11 +239,105 @@ The following activity diagram summarizes what happens when a user executes a ne
 
 _{more aspects and alternatives to be added}_
 
+### \[Proposed\] Consultation Management
+
+The consultation management feature enables TAs to schedule and manage consultation sessions with students. This section describes the implementation details of the consultation system.
+
+#### Architecture
+
+The consultation feature comprises these key components:
```
Comment (Clarenceeey) [inline docs/DeveloperGuide.md:248]: Remember to add documentation about the AddToConsultCommand (adds list of students passed to the consult), DeleteConsultCommand and RemoveFromConsultCommand (removes all students in the list passed from the consult).

**Review by Clarenceeey — COMMENTED** [review]
Add the additional functions for consults to developer guide.

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (notnotmax) [reply docs/DeveloperGuide.md:518]: > Do we also need to check if the date and time are before or after the current date and time? i.e. adding a consultation for May when it is already June.

I don't think we need to add a hard restriction for this. Maybe the TA wants to do it to record that there was a consultation that took place earlier or something. Instead, maybe the GUI can show past consultations with their dates highlighted in red?

**Comment by marcusjhang** [comment]
lgtm

**Pull request merged**